src/platform/i386/exception.c

```diff
@@ -79,9 +79,9 @@ int
 device_not_avail_fault_handler(struct pt_regs *regs)
 {
 	print_regs_state(regs);
-	die("FAULT: Device Not Available\n");
+	printk("\n\n\n\n Device not avail fault handle fpu exception \n\n\n");
```

Comment:
This exception, seems to me, is triggered after a thread wants to use fpu while it has been disabled. Though it might be good enough for now, but I think in the future we want to distinguish fpu related exceptions with other "device_not_avail" exceptions.
Moreover, the message printed is not clear. If you want to use this exception for fpu enabling, you have to remove this print since it will cause overheads.

Reply:
@WenyuanShao l don't think that a modern system has any other device that causes this exception. I think it is only FPU. We can check Linux code to verify, perhaps?